crates/wit-component/src/printing.rs

```diff
@@ -33,98 +75,122 @@ impl Default for WitPrinter {
+    /// Configure whether doc comments will be printed.
+    ///
+    /// Defaults to true.
+    pub fn emit_docs(&mut self, enabled: bool) -> &mut Self {
+        self.emit_docs = enabled;
+        self
+    }
+
+    /// Prints the specified `pkg`.
+    ///
+    /// If `is_main` is not set, nested package notation is used.
+    pub fn print_package(
+        &mut self,
+        resolve: &Resolve,
+        pkg: PackageId,
+        is_main: bool,
+    ) -> Result<()> {
         let pkg = &resolve.packages[pkg];
-        self.print_docs(&pkg.docs);
-        self.output.push_str("package ");
-        self.print_name(&pkg.name.namespace);
-        self.output.push_str(":");
-        self.print_name(&pkg.name.name);
-        if let Some(version) = &pkg.name.version {
-            self.output.push_str(&format!("@{version}"));
-        }
+        self.print_package_outer(pkg)?;
 
         if is_main {
-            self.print_semicolon();
-            self.output.push_str("\n\n");
+            self.output.semicolon();
+            self.output.newline();
         } else {
-            self.output.push_str(" {\n");
+            self.output.indent_start();
         }
 
         for (name, id) in pkg.interfaces.iter() {
-            self.print_docs(&resolve.interfaces[*id].docs);
-            self.print_stability(&resolve.interfaces[*id].stability);
-            self.output.push_str("interface ");
-            self.print_name(name);
-            self.output.push_str(" {\n");
+            self.print_interface_outer(resolve, *id, name)?;
+            self.output.indent_start();
             self.print_interface(resolve, *id)?;
+            self.output.indent_end();
             if is_main {
-                writeln!(&mut self.output, "}}\n")?;
-            } else {
-                writeln!(&mut self.output, "}}")?;
+                self.output.newline();
             }
         }
 
         for (name, id) in pkg.worlds.iter() {
             self.print_docs(&resolve.worlds[*id].docs);
             self.print_stability(&resolve.worlds[*id].stability);
-            self.output.push_str("world ");
-            self.print_name(name);
-            self.output.push_str(" {\n");
+            self.output.keyword("world");
+            self.output.str(" ");
+            self.print_name_type(name, TypeKind::WorldDeclaration);
+            self.output.indent_start();
             self.print_world(resolve, *id)?;
-            writeln!(&mut self.output, "}}")?;
+            self.output.indent_end();
         }
         if !is_main {
-            writeln!(&mut self.output, "}}")?;
+            self.output.indent_end();
         }
         Ok(())
     }
 
-    fn print_semicolon(&mut self) {
-        self.output.push_str(";");
+    /// Print the specified package without its content.
+    /// Does not print the semicolon nor starts the indentation.
+    pub fn print_package_outer(&mut self, pkg: &Package) -> Result<()> {
```

Comment:
How come some methods were made `pub` in this PR? It's totally ok to do so, but I just wanted to confirm this wasn't by accident and it was intentional

Reply:
It was intentional, answering here: https://github.com/bytecodealliance/wasm-tools/pull/1956#issuecomment-2551305004